File: apps/alfred/api/v1/wikipedia.py
from __future__ import annotations

from fastapi import APIRouter, Query

from alfred.services.wikipedia import retrieve_wikipedia

router = APIRouter(prefix="/api/v1/wikipedia", tags=["wikipedia"])


@router.get("/search")
def search(
    q: str = Query(...),
    lang: str = Query("en"),
    top_k_results: int = Query(3, ge=1, le=50),
    doc_content_chars_max: int = Query(4000, ge=200, le=20000),
    load_all_available_meta: bool = Query(False),
    load_max_docs: int = Query(100, ge=1, le=500),
):
    return retrieve_wikipedia(
        query=q,
        lang=lang,
        top_k_results=top_k_results,
        doc_content_chars_max=doc_content_chars_max,
        load_all_available_meta=load_all_available_meta,
        load_max_docs=load_max_docs,
    )

File: frontend/lib/api/wikipedia.ts
import { apiFetch } from "@/lib/api/client";

import type { WikipediaSearchResponse } from "@/lib/api/types/wikipedia";

export type WikipediaSearchParams = {
  q: string;
  lang?: string;
  top_k_results?: number;
  doc_content_chars_max?: number;
  load_all_available_meta?: boolean;
  load_max_docs?: number;
};

export async function wikipediaSearch(params: WikipediaSearchParams): Promise<WikipediaSearchResponse> {
  const query = new URLSearchParams();
  query.set("q", params.q);
  if (params.lang) query.set("lang", params.lang);
  if (typeof params.top_k_results === "number") query.set("top_k_results", String(params.top_k_results));
  if (typeof params.doc_content_chars_max === "number")
    query.set("doc_content_chars_max", String(params.doc_content_chars_max));
  if (params.load_all_available_meta) query.set("load_all_available_meta", "true");
  if (typeof params.load_max_docs === "number") query.set("load_max_docs", String(params.load_max_docs));
  return apiFetch<WikipediaSearchResponse>(`/api/wikipedia/search?${query.toString()}`, {
    cache: "no-store",
  });
}

